src/java/org/apache/cassandra/index/sai/disk/vector/CompactionGraph.java

```diff
@@ -182,6 +182,14 @@ public CompactionGraph(IndexComponents.ForWrite perIndexComponents, ProductQuant
     private OnDiskGraphIndexWriter createTermsWriter(int maxVectorOrdinal) throws IOException
     {
         var indexConfig = context.getIndexWriterConfig();
+        // We can use IdentityMapper here because it does the right thing in all three scenarios:
```

Comment:
This is a helpful comment. I didn't dig in enough to understand why the IdentityMapper worked for all cases until reading this.